pandas/_libs/tslibs/conversion.pyx

```diff
@@ -382,6 +385,14 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
             obj.tzinfo = tz
     else:
         obj.value = pydatetime_to_dt64(ts, &obj.dts)
+        # GH 24329 Take DST offset into account
+        # Two check in if necessary because class
+        # differs on Windows and Linux
+        if (isinstance(ts.tzinfo, du_tzfile1) or
+                isinstance(ts.tzinfo, du_tzfile2)):
+            if ts.tzinfo.is_ambiguous(ts):
+                dst_offset = ts.tzinfo.dst(ts)
+                obj.value += int(dst_offset.total_seconds() * 1e9)
```

Comment:
Because `pydatetime_to_dt64` doesn't take DST into account but `get_utcoffset` does, we need to add the DST timedelta to the result of `pydatetime_to_dt64`. This is necessary only for `dateutil`, because for ambiguous dates `pytz` immediately switches off DST and reduces UTC offset, so no correction is necessary for `pytz` timezones.

Reply:
can you add a comment to this effect here